staging/src/k8s.io/client-go/features/kube_features.go

```diff
@@ -41,6 +43,34 @@ const (
 	WatchList featuregate.Feature = "WatchList"
 )
 
+func DefaultFeatureGates() featuregate.FeatureGate {
+	return featureGates.Load().(featuregate.FeatureGate)
+}
+
 func init() {
-	runtime.Must(utilfeature.DefaultMutableFeatureGate.Add(defaultKubernetesFeatureGates))
+	envVarGates := featuregate.NewEnvVarFeatureGate()
+	runtime.Must(envVarGates.Add(defaultKubernetesFeatureGates))
+
+	featureGates.Store(envVarGates)
+}
+
+func AddFeaturesToExistingFeatureGates(mutableFeatureGates featuregate.MutableFeatureGate) error {
+	return mutableFeatureGates.Add(defaultKubernetesFeatureGates)
+}
+
+func SetFeatureGates(newFeatureGates featuregate.FeatureGate) {
```

Comment:
I think that it implies the following usage:

```
fg := DefaultFeatureGates().DeepCopy()
fg.Set("k=v")
SetFeatureGates(fg)
```

Which has the side effect of resetting the internal `checkEnvVarsOnce` variable. 
This means that the environment variable can be read after each `DeepCopy`.
Was this intentional?
Why ? Why not to `DefaultFeatureGates.Set("k=v")` ?